### ISSUE TITLE
old westwood-maps causing lobby-crash

### ISSUE THREAD

**Issue by Micr0Bit** [OpenRA/OpenRA#8115]
some of those old westwood maps that didnt get fixed by any players ... do allow more players then actually spots are available on the map ... this is causing the lobby to crash after a couple of seconds 

the map "stalingrad" is a example http://resource.openra.net/maps/2540/... it has 5 player-slots ... but only 4 map-starting-points

______________________________________________________________________
Red Alert Mod at Version release-20150424
Operating System: Windows (Microsoft Windows NT 6.1.7601 Service Pack 1)
Runtime Version: .NET CLR 4.0.30319.18408
Exception of type `System.ArgumentException`: Ein Element mit dem gleichen Schlüssel wurde bereits hinzugefügt.
   bei System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   bei System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   bei System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   bei System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
   bei OpenRA.Mods.Common.Widgets.Logic.LobbyUtils.GetSpawnOccupants(Session lobbyInfo, MapPreview preview)
   bei OpenRA.Mods.Common.Widgets.Logic.LobbyMapPreviewLogic.<>c__DisplayClass47.<.ctor>b__3()
   bei OpenRA.Mods.Common.Widgets.MapPreviewWidget.Draw()
   bei OpenRA.Widgets.Widget.DrawOuter()
   bei OpenRA.Widgets.Widget.DrawOuter()
   bei OpenRA.Widgets.Widget.DrawOuter()
   bei OpenRA.Widgets.Widget.DrawOuter()
   bei OpenRA.Widgets.Widget.DrawOuter()
   bei OpenRA.Widgets.Widget.DrawOuter()
   bei OpenRA.Widgets.Widget.DrawOuter()
   bei OpenRA.Widgets.Ui.Draw()
   bei OpenRA.Game.RenderTick()
   bei OpenRA.Game.Loop()
   bei OpenRA.Game.Run()
   bei OpenRA.Program.Run(String[] args)
   bei OpenRA.Program.Main(String[] args)
________________________________________________________________

**Comment by MicroJOo** [comment]
River Crossing is another one ... 7 Slots , 4 startpoints 
http://resource.openra.net/maps/2537

![lobby crash](https://cloud.githubusercontent.com/assets/9139965/7547545/132b8a06-f5f0-11e4-97b7-ddb3cb6c1dba.png)

**Comment by Mailaender** [comment]
The real problem is spawn points at the same place:

```yaml
	Actor59: mpspawn
		Location: 91,79
		Owner: Neutral
	Actor60: mpspawn
		Location: 39,41
		Owner: Neutral
	Actor61: mpspawn
		Location: 38,88
		Owner: Neutral
	Actor62: mpspawn
		Location: 88,37
		Owner: Neutral
	Actor63: mpspawn
		Location: 39,41
		Owner: Neutral
	Actor64: mpspawn
		Location: 88,37
		Owner: Neutral
	Actor65: mpspawn
		Location: 38,88
		Owner: Neutral
```

**Comment by pchote** [comment]
@Mailaender: We should also check that these duplicates aren't caused by a bug in the map importer before marking this as fixed.

**Comment by dan9550** [comment]
The main problem here (correct me if i'm wrong) the legacy maps actually had more starting positions than player slots. The player would be given a spawning point at random.
i'm guessing we will never support this behavior either.

**Comment by penev92** [comment]
I don't know about C&C and RA, but Dune 2000 maps had 8 starting positions, even the 2-player maps, and players get a random spawn pont, so you are probably right.
Although the problem here isn't exactly the same - there are more player references than spawn points.

**Comment by dan9550** [comment]
@penev92 Ahh i see, my bad this is what happens in the morning when your not a coffee drinker...

**Comment by abcdefg30** [comment]
Is this considered as fixed now as #8154 is merged, but #11240 rejected?

**Comment by Mailaender** [comment]
The legacy map import in LoadWaypoints needs to filter those duplicate spawn points.
